src/crimson/os/seastore/seastore_types.h

```diff
@@ -174,13 +174,14 @@ std::ostream &segment_to_stream(std::ostream &, const segment_id_t &t);
 
 // Offset within a segment on disk, see SegmentManager
 // may be negative for relative offsets
-using segment_off_t = int32_t;
-constexpr segment_off_t NULL_SEG_OFF =
-  std::numeric_limits<segment_off_t>::max();
-constexpr segment_off_t MAX_SEG_OFF =
-  std::numeric_limits<segment_off_t>::max();
+using seastore_off_t = int32_t;
+using device_seastore_off_t = uint32_t;
```

Comment:
What is a device_seastore_off_t?

Reply:
This is added to use 32 bit (unsigned) helpers such as ::max(). In other word, the purpose of this is to use entire 32 bit, which is the same size of the seastore_off_t,  excluding MSB.